### PR TITLE
Apply default settings per locale

### LINKED ISSUE #696: Different default settings per locale
Be able to customize the default settings (e.g. selected translations, tafsirs, font, qari, etc.) based on the locale.

### PR DESCRIPTION
**Pull request by osamasayed**
### Summary
This PR adds the ability to customize the default settings based on the locale. For example when the locale is `ar` (Arabic), we want to make the default reading mode as reading instead of translation. 

Closes #696

### CODE REVIEW

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/\[chapterId\]/\[verseId\]**](https://quran-com-3g6r02uty-qurancom.vercel.app/[chapterId]/[verseId]) | [**…rId\]/\[verseId\]/tafsirs**](https://quran-com-3g6r02uty-qurancom.vercel.app/[chapterId]/[verseId]/tafsirs) |
|:-:|:-:|
|<a href="https://quran-com-3g6r02uty-qurancom.vercel.app/[chapterId]/[verseId]">Screenshot of /[chapterId]/[verseId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/OzTp9W0PDy6yO8kJnhtbcvpPYKnJX_TZwDfb7iTRy_lvymuN_511ou9R9B4to3wcOOid3JyYXKO3BnyaxCC0UQ.png">(view full size)</a> |<a href="https://quran-com-3g6r02uty-qurancom.vercel.app/[chapterId]/[verseId]/tafsirs">Screenshot of /[chapterId]/[verseId]/tafsirs</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/4h_-sHqNT-1Adu9FBj1UcxoZuhgN7BAmNm9TFFlMSeZ9ybbkomDuN_EmabCwyBLYKA6i8wmXN1yZoVZ5gvnmjQ.png">(view full size)</a> |



| [**/\[chapterId\]**](https://quran-com-3g6r02uty-qurancom.vercel.app/[chapterId]) | [**…Id\]/tafsirs/\[tafsirId\]**](https://quran-com-3g6r02uty-qurancom.vercel.app/[chapterId]/tafsirs/[tafsirId]) |
|:-:|:-:|
|<a href="https://quran-com-3g6r02uty-qurancom.vercel.app/[chapterId]">Screenshot of /[chapterId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/NqC0YvG6zORYSr2zXn5Ggsm4-ymXmaSJ-kSrqKFhTnzUD9RF-N-MuDagfzItqdE8uP3J0RCMo-F-CT8bhXqEfQ.png">(view full size)</a> |<a href="https://quran-com-3g6r02uty-qurancom.vercel.app/[chapterId]/tafsirs/[tafsirId]">Screenshot of /[chapterId]/tafsirs/[tafsirId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Hs01bpEMg3jT40a4M_WQh720vq0Tlz4RzQobJyOPGw6Y58W_i0W6UUEk3l80yVbAgZ_FFQGIzTYR8_jMzkcGNg.png">(view full size)</a> |



| [**/juz/\[juzId\]**](https://quran-com-3g6r02uty-qurancom.vercel.app/juz/[juzId]) | [**/page/\[pageId\]**](https://quran-com-3g6r02uty-qurancom.vercel.app/page/[pageId]) |
|:-:|:-:|
|<a href="https://quran-com-3g6r02uty-qurancom.vercel.app/juz/[juzId]">Screenshot of /juz/[juzId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/DgVjAsyUGQ_OjCki9HwoxEz9K9AP5dwFe8NTYhA9p1TiT0AKvmMO0qxB13rAiMewN41Ln4q5RrSnQKZhMoUyog.png">(view full size)</a> |<a href="https://quran-com-3g6r02uty-qurancom.vercel.app/page/[pageId]">Screenshot of /page/[pageId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/1siIySf2FCn48kIEA2hSgkQdLwKkKzm73FTyILw3Ude8Uh7Zs4ol2kTvOOnKvFFolGZzrGeY0iQWSIhQRtVv4g.png">(view full size)</a> |







Commit ad39962d0a61489bb1086d6d83a607f719fbebda (https://quran-com-3g6r02uty-qurancom.vercel.app).

File: src/redux/defaultSettings/util.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+export const getTranslationsInitialState = (locale = DEFAULT_LOCALE): TranslationsSettings => {
+  return getLocaleInitialStateByKey(locale, 'translations');
+};
+
+export const getTafsirsInitialState = (locale = DEFAULT_LOCALE): TafsirsSettings => {
+  return getLocaleInitialStateByKey(locale, 'tafsirs');
+};
+export const getAudioPlayerStateInitialState = (locale = DEFAULT_LOCALE): AudioState => {
+  return getLocaleInitialStateByKey(locale, 'audioPlayerState');
+};
```
Comment (tifa2UP) [inline src/redux/defaultSettings/util.ts:76]: nice

File: src/redux/middleware/defaultSettingsMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * in turn dispatches an action that indicates that the user is
+ * no longer using the default settings which will be used later
+ * when switching between locales to determine whether we want
+ * to apply the new locale's default settings or keep the current
+ * setting as they are.
+ *
+ * @param {MiddlewareAPI<Dispatch<AnyAction>} storeAPI
+ * @returns {Dispatch<any>(action: any) => any}
+ */
+const DefaultSettingsMiddleware: Middleware<
```
Comment (tifa2UP) [inline src/redux/middleware/defaultSettingsMiddleware.ts:43]: Have some thoughts on this. Do we actually need to use a middleware? 

or can we deduce the "is using default settings" from the value of the current store. 

Let's chat about i tomorrow iA.

File: src/redux/defaultSettings/defaultSettings.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  // the base sizes in rem
+  tafsirFontScale: 3,
+  quranTextFontScale: 3,
+  translationFontScale: 3,
+  quranFont: QuranFont.QPCHafs,
+  mushafLines: MushafLines.SixteenLines,
+};
+
+const DEFAULT_TRANSLATION = 20; // just a placeholder.
+const DEFAULT_TRANSLITERATION = 12; // just a placeholder.
```
[the remark this answers is not in the document]
Comment (osamasayed) [reply src/redux/defaultSettings/defaultSettings.ts:63]: It was there from before, I will remove it

File: src/redux/defaultSettings/util.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+const DEFAULT_LOCALE = 'en';
+
+/**
+ * Dynamically load the default settings of the locale passed.
+ *
+ * @param {string} locale
+ * @returns {DefaultSettings}
+ */
+const importLocaleFile = (locale: string): DefaultSettings =>
+  require(`src/redux/defaultSettings/locales/${locale}`).default;
```
[the remark this answers is not in the document]
Comment (tifa2UP) [reply src/redux/defaultSettings/util.ts:43]: Move to a separate file, create a switch statement for all the different paths.

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/\[chapterId\]/\[verseId\]**](https://quran-com-44w6lucm0-qurancom.vercel.app/[chapterId]/[verseId]) | [**…rId\]/\[verseId\]/tafsirs**](https://quran-com-44w6lucm0-qurancom.vercel.app/[chapterId]/[verseId]/tafsirs) |
|:-:|:-:|
|<a href="https://quran-com-44w6lucm0-qurancom.vercel.app/[chapterId]/[verseId]">Screenshot of /[chapterId]/[verseId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/69IvtsZeLCBBTz6zA81okJV_RGS6BKV0iexmipDZNrun_VC2jkeAVuxcdhn8oiGMUlSh4vM9qXc08QSBuNVYXg.png">(view full size)</a> |<a href="https://quran-com-44w6lucm0-qurancom.vercel.app/[chapterId]/[verseId]/tafsirs">Screenshot of /[chapterId]/[verseId]/tafsirs</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/hSuGfLeuCEtV-X5karATOPGMbymWI-SGYecMhEwSO_KH9D0MKGUsDCsiidS-kZkKgB_2M_r313xloSmLUFJNPQ.png">(view full size)</a> |



| [**/\[chapterId\]**](https://quran-com-44w6lucm0-qurancom.vercel.app/[chapterId]) | [**…Id\]/tafsirs/\[tafsirId\]**](https://quran-com-44w6lucm0-qurancom.vercel.app/[chapterId]/tafsirs/[tafsirId]) |
|:-:|:-:|
|<a href="https://quran-com-44w6lucm0-qurancom.vercel.app/[chapterId]">Screenshot of /[chapterId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/xYW-2KOGxo2VejjZ19j4VG_Hpb-UoOTuNJehZ1M-dBGYpEMmpAr3Mo1g8qbBIwOL4HEvigPHb9jK6APD8I85tg.png">(view full size)</a> |<a href="https://quran-com-44w6lucm0-qurancom.vercel.app/[chapterId]/tafsirs/[tafsirId]">Screenshot of /[chapterId]/tafsirs/[tafsirId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/rfKOuNfiNRbPzW7jcU-wFtCmgR7jl-qf19taguGz7f9ZO0gYi-_bQ0e3xB4l98FKJmhkotF5Dzx9E6sgjKEKcQ.png">(view full size)</a> |



| [**/juz/\[juzId\]**](https://quran-com-44w6lucm0-qurancom.vercel.app/juz/[juzId]) | [**/page/\[pageId\]**](https://quran-com-44w6lucm0-qurancom.vercel.app/page/[pageId]) |
|:-:|:-:|
|<a href="https://quran-com-44w6lucm0-qurancom.vercel.app/juz/[juzId]">Screenshot of /juz/[juzId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/G6awFGJweHPPtVuLQPQR4q20vxD5OulUhYVV8zPOOc1j1phpMGbINHVdyKjL5uqB3V6y_JtLQCepyuJIcT8-xg.png">(view full size)</a> |<a href="https://quran-com-44w6lucm0-qurancom.vercel.app/page/[pageId]">Screenshot of /page/[pageId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/XSyh8lWZasQF5P8fhheQ_U6hhh4ytPgNiTivCXAaz_-YYv57gGhacgWc1BAOah-18QS_ja-otUnyIfn6TW5sQA.png">(view full size)</a> |







Commit 8a2abfd4d4a70539041df6aae73e98ff9848f0ca (https://quran-com-44w6lucm0-qurancom.vercel.app).

**Pull request merged**